### PR TITLE
GAWB-2820: Custom Data Access Instructions

### PR DESCRIPTION
**Pull request by MattPutnam**
[Corresponding orch PR](https://github.com/broadinstitute/firecloud-orchestration/pull/538)

* Made MarkdownEditor managed or unmanaged
* Respond to 'markdown' render hint in catalog wizard
* Render instructions if present

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [x] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [x] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by davidangb** [comment]
in the catalog wizard, the side-by-side view has the markdown (right) side totally scrunched up with a width of 2-3 characters. I'd be fine disabling side-by-side here given the size constraints.

File: src/cljs/main/broadfcui/common/markdown.cljs [hunk 1]
```diff
@@ -32,29 +32,37 @@
 
 (react/defc MarkdownEditor
   {:get-trimmed-text
-   (fn [{:keys [state]}]
+   (fn [{:keys [state this]}]
+     (assert (not (this :-managed?)) "Trying to read text from a managed component")
```
Comment (aednichols) [inline src/cljs/main/broadfcui/common/markdown.cljs]: I'm confused by what `:-managed?` does. I thought maybe it was a React thing, but I can't find anything in the docs

**Review by aednichols — COMMENTED** [review]
Code looks reasonable, waiting for dev to be back to play around with it

File: src/cljs/main/broadfcui/common/markdown.cljs (hunk 1 above)
Comment (MattPutnam) [reply src/cljs/main/broadfcui/common/markdown.cljs]: Hmm, proper language appears to be "controlled", though I swear I've heard "managed". I'll update it. Anyway: https://reactjs.org/docs/uncontrolled-components.html

**Comment by aednichols** [comment]
jenkins retest

File: src/cljs/main/broadfcui/common/markdown.cljs [hunk 2]
```diff
@@ -32,29 +32,37 @@
...
+                                                             ((:on-change props) new-value)
+                                                             (swap! state assoc :text new-value)))
                                               :style {:width "100%" :height "100%" :minHeight 200
                                                       :resize (if (= mode :edit) "vertical" "none")}})]
        [:div {}
-        [:div {}
+        [:div {:style {:marginTop "0.5rem"}}
          (tab :edit "Edit")
          (tab :preview "Preview")
          (tab :side-by-side "Side-by-side")]
```
Comment (zarsky-broad) [inline src/cljs/main/broadfcui/common/markdown.cljs:69]: I don't know if it was introduced here or earlier, but the buttons seem to be overlapped by the view in edit and side-by-side (try making the preview side wide enough that it starts before the buttons end)

Comment (MattPutnam) [reply src/cljs/main/broadfcui/common/markdown.cljs:69]: Looks like it's always been like that, but I fixed it.

**Review by aednichols — APPROVED** [review]
Tested that a markdown message shows up after setting one for a test dataset.

**Review by davidangb — APPROVED** [review]
:+1: working for me!

**Review by zarsky-broad — APPROVED** [review]

**Pull request merged**